theory/theory.md

```diff
@@ -0,0 +1,362 @@
+
+2. Security
+
+POP is not as secure as OOPs. Because of abstraction, data hiding is possible in object-oriented programming. 
+
+3. Approach
+
+POP is approached from the top down. OOP is built from the ground up.
+
+4. Data movement
+
+Data travels freely within the system from one function to the next in Process Oriented Programmi. Objects in OOP can move and communicate with one another using member functions.
+
+5. Orientation
+
+POP is structured/procedural in nature whereas OOP is object-oriented in nature.
+
+6. Access modifiers	
+
+In POP, there are no access modifiers. In OOP, access modifiers are classified as private, public, or protected.
+
+7. Inheritance	
+
+There is a feature of inheritance in OOP only. 
+
+8. Code reusability	
+
+In POP, there is no code reusability. OOP provides code reusability by utilising the inheritance capability.
+
+9. Overloading
+
+Function overloading and operator overloading are concepts only in OOP.
+
+10. Importance
+
+OOP gives importance to data over functions.
+
+11. Virtual class	
+
+In OOP, there could be virtual classes in inheritance.
+
+12. Complex problems	
+
+OOP is more appropriate for complex systems.
+
+13. Data hiding	
+
+It is possible to hide data in OPP. 
+
+14. Program division	
+
+In OOP, a program is made of small parts that are referred to as objects, not lfunction like POP. 
+
+15. languages Examples	
+
+Process Oriented Programming languages : C, Fortran, Pascal, and VB
+Object-oriented programming languages : .NET, C#, Python, Java, VB.NET, and C++.
+
+Resource: https://www.javatpoint.com/procedural-programming-vs-object-oriented-programming	
+
+
+## 2. What's polymorphism in OOP?
+
+The meaning of ‘polymorphism’ is the condition of occurrence in different forms.
+
+Polymorphism is a concept in OOP, which refers to the use of a single type entity (method, operator or object) to represent different types in different scenarios.
+
+Polymorphism appears in two type:Static and dynamic polymorphism
+
+Here is an example of polymorphism with Classes: 
+
+    class Adult:
+        def move(self):
+            print('walk')
+    
+        def greet(self):
+            print('Hello!')
+
+
+    class Baby:
+        def move(self):
+            print('crawl')
+    
+        def greet(self):
+            print('smile :)')
+
+
+    Keiko = Adult()
+    Emma = Baby()
+    
+    for human in (Keiko, Emma):
+        human.move()
+        human.greet()
+    
+    # output
+    # walk
+    # Hello!
+    # crawl
+    # smile :)
+
+Here, we have created two classes Adultand Baby. They share a similar structure and have the same method names move() and greet(). Although we have not created a common superclass or linked the classes together, we can pack these two different objects into a tuple and iterate through it using a common human variable. It is possible due to polymorphism. 
+
+
+Source: https://www.programiz.com/python-programming/polymorphism
+

```

Comment:
Q2. 8 / 10
For this question, you have done a great job at explain the important details. However, certain terminologies were never mentioned. Expected concepts like inheritance, overriding as it pertains polymorphism to be explained within.